addons/mail/controllers/webclient.py

```diff
@@ -104,3 +104,12 @@ def _process_request_for_internal_user(self, store: Store, name, params):
                 ("group_ids", "in", request.env.user.all_group_ids.ids),
             ]
             store.add(request.env["mail.canned.response"].search(domain))
+        if name == "avatar_card":
+            if (
+                user := request.env["res.users"]
+                .with_context(active_test=False)
+                .search([("id", "=", params["user_id"])])
+            ):
+                company_ids = list(request.env.user._get_company_ids())
+                user = user.with_context(allowed_company_ids=company_ids)
```

Comment:
There is already a `with_context` line above, this could be moved into the same one.
Not only that, I'm not sure but there might be value to have the correct companies before doing the search.